File: game_client/Assets/Scripts/ResScripts/NPResCommon/PlayerCondition/ConditionObj/NPPlayerCondition_CS_MARS_TECH_LVL.cs
using NPEnum;
using ALPackage;

namespace GOE
{
    /// <summary>
    /// 火星科技等级 CS_MARS_TECH_LVL:tech_id(:minLevel:maxLevel)
    /// </summary>
    public class NPPlayerCondition_CS_MARS_TECH_LVL : _ANPBasicPlayerCondition
    {
        public override ENPPlayerConditionType conditionType { get { return ENPPlayerConditionType.CS_MARS_TECH_LVL; } }

        private long _m_lTechId;
        private WCGLongRange _m_lRange;

        public static NPPlayerCondition_CS_MARS_TECH_LVL readStr(ALStringReader _reader)
        {
            string techId = _reader.readItem(':');
            if (null == techId)
            {
                UnityEngine.Debug.LogError("Can not read str for ENPPlayerConditionType.CS_MARS_TECH_LVL[" + _reader.srcString + "]");
                return null;
            }

            NPPlayerCondition_CS_MARS_TECH_LVL cond = new NPPlayerCondition_CS_MARS_TECH_LVL();

            cond._m_lTechId = ALCommon.ParseLong(techId);

            string minVS = _reader.readItem(':');
            if (null != minVS)
            {
                string maxVS = _reader.readItem(':');
                if (null != maxVS)
                    cond._m_lRange = new WCGLongRange(minVS, maxVS);
                else
                    cond._m_lRange = new WCGLongRange(minVS, "-1");
            }
            else
            {
                cond._m_lRange = new WCGLongRange(1, -1);
            }

            return cond;
        }

        public override bool isEnable(NPVarInfo _varVariableInfo)
        {
#if NP_GAME
            if (NPPlayer.instance == null || NPPlayer.instance.marsComp == null)
                return false;

            MarsTechnologyInfo techInfo = NPPlayer.instance.marsComp.technologySubComponent.getTechnologyInfoById(_m_lTechId);
            if (techInfo == null)
                return false;

            long level = techInfo.lvl;
            return isRange(level, _m_lRange.min, _m_lRange.max);
#else
            return false;
#endif
        }
    }
}

File: game_server/GameRes/src/NPGameRes/GameObjs/PlayerCondition/ConditionObj/NPPlayerCondition_CS_MARS_TECH_LVL.java
package NPGameRes.GameObjs.PlayerCondition.ConditionObj;

import ALServerLog.ALServerLog;
import NPCommon.CommonObj.NPStringReader;
import NPEnum.ENPPlayerConditionType;
import NPGameRes.GameObjs.PlayerCondition._ANPBasicPlayerCondition;

/**
 * 火星指定科技等级判断 CS_MARS_TECH_LVL:科技id:min_value（:max_value）
 * @author mark
 */
public class NPPlayerCondition_CS_MARS_TECH_LVL extends _ANPBasicPlayerCondition
{
    private long _m_lId;
    private long _m_lMinValue; // -1:不限制最小值
    private long _m_lMaxValue = -1; //-1:不限制最大值

    public long id() {return _m_lId;}
    public long minValue() {return _m_lMinValue;}
    public long maxValue() {return _m_lMaxValue;}

    @Override
    public ENPPlayerConditionType conditionType()
    {
        return ENPPlayerConditionType.CS_MARS_TECH_LVL;
    }

    public static NPPlayerCondition_CS_MARS_TECH_LVL readStr(NPStringReader _reader)
    {
        NPPlayerCondition_CS_MARS_TECH_LVL cond = new NPPlayerCondition_CS_MARS_TECH_LVL();

        String idS = _reader.readItem();
        String minVS = _reader.readItem();

        if (null == idS || null == minVS)
        {
            ALServerLog.Error("Can not read str for ENPPlayerConditionType.CS_MARS_TECH_LVL[" + _reader.getSrcString() + "]");
            return null;
        }

        cond._m_lId = Long.parseLong(idS);
        cond._m_lMinValue = Long.parseLong(minVS);

        String maxVS = _reader.readItem();
        if (null != maxVS)
        {
        	cond._m_lMaxValue = Long.parseLong(maxVS);
        }

        return cond;
    }
}
